### PR TITLE
add options to filter dependencies by kind; defaults to Kind::Normal

### LINKED ISSUE #42: Switch to ignore build-dependencies and dev-dependencies
Although ideally build code should be all safe, often build and dev dependencies pull in a lot of other crates containing unsafe. There are generally less useful to examine than the code going into the crate, and it would be useful to be able to ignore them.

Further, i would suggest ignoring build and dev dependencies could be done by default, with the flag being there to add them back into analysis for the cases where you need them.

I propose the flags should be `--all-dependencies | --all-deps`, that would alias `--build-dependencies | -- build-deps` and `--dev-dependencies | --dev-deps`. As the names suggest, these would include all, just the build, or just the dev dependencies, _in addition_ to, and not instead, of the crate and regular dependency code.

### PR DESCRIPTION
**Pull request by alexmaco**
fixes #42 

Let me know if any more documentation and/or help text is needed (in which cases, suggestions would be appreciated)

### CODE REVIEW

**Comment by anderejd** [comment]
Thank you!

I will find some time to review and merge tomorrow evening.

File: src/main.rs [hunk 1]
```diff
@@ -132,6 +132,18 @@ pub struct Args {
     #[structopt(long = "include-tests")]
     /// Count unsafe usage in tests.
     pub include_tests: bool,
+
+    #[structopt(long = "build-dependencies", alias = "build-deps")]
+    /// Also analize build dependencies
```
Comment (anderejd) [inline src/main.rs]:
```suggestion
    /// Also analyze build dependencies
```

File: src/main.rs [hunk 2]
```diff
@@ -132,6 +132,18 @@ pub struct Args {
...
+    pub build_deps: bool,
+
+    #[structopt(long = "dev-dependencies", alias = "dev-deps")]
+    /// Also analize dev dependencies
```
Comment (anderejd) [inline src/main.rs]:
```suggestion
    /// Also analyze dev dependencies
```

File: src/main.rs [hunk 3]
```diff
@@ -132,6 +132,18 @@ pub struct Args {
...
+    pub dev_deps: bool,
+
+    #[structopt(long = "all-dependencies", alias = "all-deps")]
+    /// Analize all depencencies, including build and dev
```
Comment (anderejd) [inline src/main.rs]:
```suggestion
    /// Analyze all dependencies, including build and dev
```

**Comment by alexmaco** [comment]
I couldn't find a way to commit suggested changes as fixups, so i force-pushed after squashing them together locally.

**Comment by anderejd** [comment]
LGTM!

**Pull request merged**